test/UI/page_object/forms_page.py

```diff
@@ -1,61 +1,135 @@
+    EMAIL_FIELD = (By.CSS_SELECTOR, "#userEmail")
+    HOBBY_CHECKBOX = (By.XPATH, "//label[contains(@for,'hobbies-checkbox-') and contains(text(), '{}')]")
+    CURRENT_ADDRESS_FIELD = (By.XPATH, "//textarea[@placeholder='Current Address']")
+
+    WINDOW_OF_SUBMITTED_DATA = (By.XPATH, "//div[text()='Thanks for submitting the form']")
 
     # Student Registration Form
-    def click_practice_form(self):
-        self.click(forms_page_locators.PRACTICE_FORM)
-
-    def fill_first_name(self, first_name):
-        self.send_keys(forms_page_locators.FIRST_NAME_FIELD, first_name)
-
-    def fill_last_name(self, last_name):
-        self.send_keys(forms_page_locators.LAST_NAME_FIELD, last_name)
-
-    def enter_mobile_phone(self, phone):
-        self.send_keys(forms_page_locators.PHONE_FIELD, phone)
-
-    def click_gender_radio_button(self, gender):
-        self.click((By.XPATH, "//div[@id='genterWrapper']//input[@value='{}']/..".format(gender)))
-
-    def click_submit_button(self):
+    def registrate_user(self, user: User):
+        self.enter_first_name(user.first_name)
+        self.enter_last_name(user.last_name)
+        self.enter_email(user.email)
+        self.click_gender_radio_button(user.gender)
+        self.enter_mobile_phone(user.phone)
+        self.enter_date_of_birth(user.date_of_birth)
+        self.enter_subjects(user.subjects)
+        self.click_hobby(user.hobbies)
+        self.enter_current_address(user.current_address)
+        self.enter_state(user.state)
+        self.enter_city(user.state)
+
+    def click_practice_form(self) -> None:
+        self.click(self.PRACTICE_FORM)
+
+    def enter_first_name(self, first_name: str) -> None:
+        self.send_keys(self.FIRST_NAME_FIELD, first_name)
+
+    def enter_last_name(self, last_name: str) -> None:
+        self.send_keys(self.LAST_NAME_FIELD, last_name)
+
+    @handle_none_argument
+    def enter_email(self, email: str) -> None:
+        self.send_keys(self.EMAIL_FIELD, email)
+
+    def click_gender_radio_button(self, gender: str) -> None:
+        self.click(self.replace_placeholders_in_locator(self.GENDER_RADIO_BUTTON, gender))
+
+    def enter_mobile_phone(self, phone: str) -> None:
+        self.send_keys(self.PHONE_FIELD, phone)
+
+    @handle_none_argument
+    def enter_date_of_birth(self, date_of_birth: str) -> None:
+        self.send_keys(self.DATE_OF_BIRTH_FIELD, date_of_birth)
+
+    @handle_none_argument
+    def enter_subjects(self, subjects: str) -> None:
+        self.send_keys(self.SUBJECTS_FIELD, subjects)
+
+    @handle_none_argument
+    def click_hobby(self, hobbies: List[Hobby]) -> None:
+        for hobby in hobbies:
+            self.click(self.replace_placeholders_in_locator(self.HOBBY_CHECKBOX, hobby))
+
+    @handle_none_argument
+    def enter_current_address(self, address: str) -> None:
+        self.send_keys(self.CURRENT_ADDRESS_FIELD, address)
+
+    @handle_none_argument
+    def enter_state(self, state) -> None:
+        pass
```

Comment:
If this method does nothing, it's better not to write it at all. Another option is to throw exception `method not implemented yet`, if other tester will write a test and call your method assuming it set State or City.

```suggestion
```

Reply:
Added step implementation